app/src/main/java/unipd/se18/ocrcamera/MainApplication.java

```diff
@@ -0,0 +1,96 @@
+package com.example.mattia.fotocamera;
+
+import android.app.Application;
+import android.content.res.AssetManager;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class MainApplication extends Application {
+
+    public static MainApplication instance = null;
+
+    public void onCreate(){
+        super.onCreate();
+        //start copy file here
+        instance=this;
+        copyTessDataForTextRecognizor();
+    }
+
+    /**
+   	* @author Fasan Giovanni
+    * @param none
+   	* @return String of tess's data path
+    */
+
+    private String tessDataPath(){
+        return MainApplication.instance.getExternalFilesDir(null)+"/tessdata/";
+    }
+
+    /**
+    * @author Fasan Giovanni
+    * @param none
+    * @return String of tess's data parent directory
+    */
+
+    public String getTessDataParentDirectory(){
+        return MainApplication.instance.getExternalFilesDir(null).getAbsolutePath();
+    }
+
+    /**
```

Comment:
For me that i never used Tesseract was really difficult to understand every method of this library.
Could be better to use comments inside the code when you use method of this library so would be easier to me to understabnd

Reply:
As we have discarded Tesseract, I preferred to focus on something else and I have not properly commented on the code.
For this I apologize.